<?xml version="1.0"?>
<launch>
  <arg name="transport" default="udp"/>
  <arg name="scanner_ip" default="10.0.10.76"/>
  <arg name="device" default="R2300"/>
  <arg name="start_angle" default="-450000"/>
  <arg name="max_num_points_scan" default="0"/>
  <arg name="packet_type" default="C1"/>
  <arg name="scan_topic" default="/cloud"/>
  <arg name="frame_id" default="scanner"/>
  <arg name="watchdogtimeout" default="0"/>
  <arg name="watchdog" default="false"/>

  <include file="$(find pf_description)/launch/pf_bringup.launch" >
    <arg name="scanner" value="r2300" />
  </include>

  <!-- R2300 Driver -->
  <node pkg="pf_driver" type="ros_main" name="r2300_node" output="screen">
    <param name="transport" type="string" value="$(arg transport)"/>
    <param name="scanner_ip" type="string" value="$(arg scanner_ip)"/>
    <param name="port" type="string" value="0"/>
    <param name="device" type="string" value="$(arg device)"/>
    <param name="start_angle" type="int" value="$(arg start_angle)"/>
    <param name="max_num_points_scan" type="int" value="$(arg max_num_points_scan)"/>
    <param name="packet_type" type="string" value="$(arg packet_type)"/>
    <param name="scan_topic" type="string" value="$(arg scan_topic)"/>
    <param name="frame_id" type="string" value="$(arg frame_id)"/>
    <param name="watchdogtimeout" type="int" value="$(arg watchdogtimeout)"/>
    <param name="watchdog" type="bool" value="$(arg watchdog)"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="ring1" args="0 0 0 0 0.0785 0 base_link scanner_1 10" />
	<node pkg="tf" type="static_transform_publisher" name="ring2" args="0 0 0 0 0.0262 0 base_link scanner_2 10" />
	<node pkg="tf" type="static_transform_publisher" name="ring3" args="0 0 0 0 -0.0785 0 base_link scanner_3 10" />
	<node pkg="tf" type="static_transform_publisher" name="ring4" args="0 0 0 0 -0.0262 0 base_link scanner_4 10" />
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find pf_driver)/rviz/r2300.rviz" />
</launch>